src/strategy.ts

```diff
@@ -104,7 +104,11 @@ export abstract class AbstractStrategy extends PassportStrategy {
             }
 
             // Log out the current user no matter if we can verify the logged in user === logout requested user
-            req.logout();
+            req.logout((err) => {
```

Comment:
This really should block until the callback happens. The reason for this is that otherwise the code will complete running before `logout` actually is confirmed to happen or have errored. IIRC, your previous promise-based solution was better. For example, when you run the tests, you should see [this code](https://github.com/node-saml/passport-saml/blob/ef328daba4e9322346492c7cbfb3dbc1ccf2a13b/test/capturedSamlRequests.spec.ts#L929-L940) execute before the test finishes.

Reply:
Updated the PR to wait for the logout.

Reply:
This change breaks the compatibility with 0.5 because it will now require req.logout(...) to call the promise executor callbacks, otherwise the promise will never settle.